### PR TITLE
feat: phase2-cli merge command

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**
The phase2-cli `merge` command merges large and small param files.

The final phase2 participant generates small-raw params, which are converted into non-raw using `phase2 convert`. The final small-nonraw params are then merged into the large initial params to produce the final `MPCParameters` file for a circuit.

### CODE REVIEW

File: filecoin-proofs/src/bin/phase2.rs [hunk 1]
```diff
@@ -998,6 +1008,213 @@ fn verify_contribution(path_before: &str, path_after: &str, participant_contrib:
...
+            .field("h_last", &self.h_last)
+            .field("l_len", &self.l_len)
+            .field("l_first", &self.l_first)
+            .field("l_last", &self.l_last)
+            .field("cs_hash", &hex_string(&self.cs_hash))
+            .field("contributions_len_offset", &self.contributions_len_offset)
+            .field("contributions_len", &self.contributions_len)
+            .finish()
+    }
+}
```
Comment (porcuquine) [inline filecoin-proofs/src/bin/phase2.rs:1093]: Out of curiosity, what's the benefit of manually implementing vs deriving `Debug`?

Comment (DrPeterVanNostrand) [reply filecoin-proofs/src/bin/phase2.rs:1093]: ^^^ Yup

File: filecoin-proofs/src/bin/phase2.rs [hunk 2]
```diff
@@ -998,6 +1008,213 @@ fn verify_contribution(path_before: &str, path_after: &str, participant_contrib:
...
+impl FileInfo {
+    fn parse_small(path: &str) -> Self {
+        let mut file = File::open(path).expect("failed to open file");
+
+        let delta_g1 = read_g1(&mut file).expect("failed to read delta_g1");
+        let delta_g2 = read_g2(&mut file).expect("failed to read delta_g2");
+
+        let h_len_offset = G1_SIZE + G2_SIZE;
+        let h_len = file.read_u32::<BigEndian>().expect("failed to read h_len") as u64;
+        let h_offset = h_len_offset + 4;
```
[the remark this answers is not in the document]
Comment (DrPeterVanNostrand) [reply filecoin-proofs/src/bin/phase2.rs:1104]: Yea, that's a better idea. I'll make that change

**Comment by DrPeterVanNostrand** [comment]
@vmx Thanks, I added the PR description to the commit message.

> PR: #1244

Yup, you're right. There is dead code.

I have one more phase2-cli pr for a command `phase2 parse <path>`, which just parses a params file into `FileInfo` and prints it. Which I've been using to verify that small and large params have the same values for delta_g1/g2, h/l_len, h/l_first, h/l_last

**Comment by vmx** [comment]
@DrPeterVanNostrand This PR needs a rebase. I guess the rest of it is then ready to go (i'd merge this one first, so that you don't have to rebase yet another time, i'll rebase my "split-keys" PR after this on is merged).

**Pull request merged**